src/index.js

```diff
@@ -49,9 +50,13 @@ class CID {
    * new CID(<cid.buffer>)
    * new CID(<multihash>)
    * new CID(<bs58 encoded multihash>)
+   * new CID(<CID>)
    *
    */
   constructor (version, codec, multihash) {
+    if (CID.isCID(version)) {
+      return version
```

Comment:
Almost. You should not do `return <obj>` inside as constructor.

You want something like:
```
  constructor (version, codec, multihash) {
    if (CID.isCID(version)) {
      this.version = cid.version
      this.codec = cid.codec
      this.multihash = Buffer.from(cid.multihash)
      return
    }
```